Source/WebCore/platform/graphics/texmap/FloatPolygon3D.cpp

```diff
@@ -49,14 +51,38 @@ FloatPolygon3D::FloatPolygon3D(const FloatRect& rect, const TransformationMatrix
         m_normal = edge1.cross(edge2);
     }
     m_normal.normalize();
+
+    computeBounds();
 }
 
 FloatPolygon3D::FloatPolygon3D(Vector<FloatPoint3D>&& vertices, const FloatPoint3D& normal)
     : m_vertices(WTFMove(vertices))
     , m_normal(normal)
 {
+    computeBounds();
 }
 
+void FloatPolygon3D::computeBounds()
+{
+    FloatPoint3D minCorner = vertexAt(0);
+    FloatPoint3D maxCorner = vertexAt(0);
+
+    unsigned numVertices = numberOfVertices();
+    for (unsigned i = 1; i < numVertices; i++) {
+        auto point = vertexAt(i);
+        minCorner.setX(std::min(minCorner.x(), point.x()));
+        minCorner.setY(std::min(minCorner.y(), point.y()));
+        minCorner.setZ(std::min(minCorner.z(), point.z()));
+
+        maxCorner.setX(std::max(maxCorner.x(), point.x()));
+        maxCorner.setY(std::max(maxCorner.y(), point.y()));
+        maxCorner.setZ(std::max(maxCorner.z(), point.z()));
```

Comment:
IIUC, you don't use these min of Y and Z. Can you remove them?

Reply:
Then bounds wouldn't be bounding box. I think it's better to leave them so that bounds is actually the bounding box of 3d polygon. It's easier to understand. Otherwise function name should be changed to xRange or something like that